orchestration/src/main/java/com/sap/ai/sdk/orchestration/client/model/LLMModuleResult.java

```diff
@@ -12,470 +12,15 @@
-                            new LLMModuleResult()
-                                .id(id)
-                                ._object(_object)
-                                .created(created)
-                                .model(model)
-                                .choices(choices)
-                                .usage(usage);
-  }
-
-  /** Builder helper class. */
-  public interface Builder {
-    /**
-     * Set the id of this {@link LLMModuleResult} instance.
-     *
-     * @param id ID of the response
-     * @return The LLMModuleResult builder.
-     */
-    Builder1 id(@Nonnull final String id);
-  }
-
-  /** Builder helper class. */
-  public interface Builder1 {
-    /**
-     * Set the _object of this {@link LLMModuleResult} instance.
-     *
-     * @param _object Object type
-     * @return The LLMModuleResult builder.
-     */
-    Builder2 _object(@Nonnull final String _object);
-  }
-
-  /** Builder helper class. */
-  public interface Builder2 {
-    /**
-     * Set the created of this {@link LLMModuleResult} instance.
-     *
-     * @param created Unix timestamp
-     * @return The LLMModuleResult builder.
-     */
-    Builder3 created(@Nonnull final Integer created);
-  }
-
-  /** Builder helper class. */
-  public interface Builder3 {
-    /**
-     * Set the model of this {@link LLMModuleResult} instance.
-     *
-     * @param model Model name
-     * @return The LLMModuleResult builder.
-     */
-    Builder4 model(@Nonnull final String model);
-  }
-
-  /** Builder helper class. */
-  public interface Builder4 {
-    /**
-     * Set the choices of this {@link LLMModuleResult} instance.
-     *
-     * @param choices Choices
-     * @return The LLMModuleResult builder.
-     */
-    Builder5 choices(@Nonnull final List<LLMChoice> choices);
-
-    /**
-     * Set the choices of this {@link LLMModuleResult} instance.
-     *
-     * @param choices Choices
-     * @return The LLMModuleResult builder.
-     */
-    default Builder5 choices(@Nonnull final LLMChoice... choices) {
-      return choices(Arrays.asList(choices));
-    }
-  }
-
-  /** Builder helper class. */
-  public interface Builder5 {
-    /**
-     * Set the usage of this {@link LLMModuleResult} instance.
-     *
-     * @param usage The usage of this {@link LLMModuleResult}
-     * @return The LLMModuleResult instance.
-     */
-    LLMModuleResult usage(@Nonnull final TokenUsage usage);
-  }
-}
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+// ignore manually set , it will be automatically generated by Jackson during
+// serialization
+// allows the  to be set during deserialization
+@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = LLMModuleResultStreaming.class, name = "LLMModuleResultStreaming"),
+  @JsonSubTypes.Type(value = LLMModuleResultSynchronous.class, name = "LLMModuleResultSynchronous")
+})
+public interface LLMModuleResult {}
```

Comment:
(Question)

This is hand-crafted right?
Why not use deduction?

Reply:
> This is hand-crafted right?

_"yes"_

> Why not use deduction?

_"Deduction does not look recursively for discriminating field names. But that would be possible here."_